### ISSUE TITLE
Exporting to CSV after python parse

### ISSUE THREAD

**Issue by BlvdJoe** [datamade/usaddress#105]
First of all - Love this tremendously. Second, I've been trying to use the usaddress.py + csv.py + pandas to import from a csv, parse the rows, then export the rows back out into a csv. 

However, when I run the code listed below, it (1) doesn't parse the column headers correctly and (2) doesn't then parse the fields correctly. This approached was recommended to me [over on StackOverflow] (https://stackoverflow.com/questions/31798061/output-a-changing-ordereddict-to-csv), so please let me know if (1) This approach is silly and/or (2) I'm merely suffering from noob-itis as I am, in fact, terribly new to python.

    import pandas as pd
    import usaddress
    from pandas import DataFrame

    data = ['Robie House, 5757 South Woodlawn Avenue, Chicago, IL 60637',
        'State & Lake, Chicago']

    tagged_addresses = [usaddress.parse(line) for line in data]

    address_df = pd.DataFrame(tagged_addresses)

    address_df.to_csv('moooutput.csv')

![image](https://cloud.githubusercontent.com/assets/13680281/9115087/5e4516de-3c2b-11e5-91c8-87a8ae38f0e8.png)

**Comment by cathydeng** [comment]
hey @BlvdJoe!

for what you're trying to do, the tag method would make more sense than the parse method. also, great timing with this question because I just wrote some code to do this task yesterday:

```
import csvkit
import usaddress

# expected format in input.csv: first column 'id', second column 'address'
with open('input.csv', 'rU') as f:
    reader = csvkit.DictReader(f)

    all_rows = []
    for row in reader:
        try:
            parsed_addr = usaddress.tag(row['address'])
            row_dict = parsed_addr[0]
        except:
            row_dict = {'error':'True'}
            
        row_dict['id'] = row['id']
        all_rows.append(row_dict)

field_list = ['id','AddressNumber', 'AddressNumberPrefix', 'AddressNumberSuffix', 'BuildingName', 
              'CornerOf','IntersectionSeparator','LandmarkName','NotAddress','OccupancyType',
              'OccupancyIdentifier','PlaceName','Recipient','StateName','StreetName',
              'StreetNamePreDirectional','StreetNamePreModifier','StreetNamePreType',
              'StreetNamePostDirectional','StreetNamePostModifier','StreetNamePostType',
              'SubaddressIdentifier','SubaddressType','USPSBoxGroupID','USPSBoxGroupType',
              'USPSBoxID','USPSBoxType','ZipCode', 'error']

with open('output.csv', 'wb') as outfile:
    writer = csvkit.DictWriter(outfile, field_list)
    writer.writeheader()
    writer.writerows(all_rows)
```

I just used csvkit to read from & write to csv. for the output, I also added an error column to record when addresses encounter a RepeatedLabelError.

**Comment by BlvdJoe** [comment]
@cathydeng Thank you! I encountered two problems however.<br>Even after I save a csv to UTF-8 (through NotePad), I still get    "KeyError: 'id'. I can get around this problem by using try..except..., but then I lose the original ID values from the original csv file. That's not "terrible," except it prevents me from easily tracking back to the records that returned an error on the parse.<br>
```
try:
    row_dict['id'] = row['id']
except KeyError:
    pass
all_rows.append(row_dict)
```
Finally just a heads up and it may already be covered elsewhere so feel free disregard:<br>"&" characters are treated like intersections, even if they're in a string like "1 M & M Savings Building." Again, not a big deal but I did update the code to include "Secondxxxx" headers to accommodate for these false parses
